src/cider/exec/processor/DefaultBatchProcessor.cpp

```diff
@@ -19,41 +19,54 @@
  * under the License.
  */
 
-#include "DefaultBatchProcessor.h"
-#include "StatefulProcessor.h"
-#include "StatelessProcessor.h"
+#include "exec/processor/DefaultBatchProcessor.h"
+
+#include <memory>
+
 #include "cider/CiderException.h"
+#include "exec/plan/parser/SubstraitToRelAlgExecutionUnit.h"
+#include "exec/processor/StatefulProcessor.h"
+#include "exec/processor/StatelessProcessor.h"
 
-namespace cider::processor {
+namespace cider::exec::processor {
 
 DefaultBatchProcessor::DefaultBatchProcessor(const plan::SubstraitPlanPtr& plan,
                                              const BatchProcessorContextPtr& context)
     : plan_(plan), context_(context) {
   auto allocator = context->allocator();
-  const auto substraitPlan = plan_->getPlan();
   if (plan_->hasJoinRel()) {
     // TODO: currently we can't distinguish the joinRel is either a hashJoin rel
     // or a mergeJoin rel, just hard-code as HashJoinHandler for now and will refactor to
     // initialize joinHandler accordingly once the
     joinHandler_ = std::make_shared<HashProbeHandler>(shared_from_this());
     this->state_ = BatchProcessorState::kWaiting;
   }
-  // TODO: compile substrait plan
+
+  auto translator =
+      std::make_shared<generator::SubstraitToRelAlgExecutionUnit>(plan_->getPlan());
+  RelAlgExecutionUnit ra_exe_unit = translator->createRelAlgExecutionUnit();
+  jitlib::CompilationOptions co;
+  co.dump_ir = true;
```

Comment:
shouldn't set this by default.

Reply:
will be removed after all integration done, now just to make sure it call nextgen.